src/xrpld/rpc/detail/TransactionSign.cpp

```diff
@@ -768,38 +872,11 @@ checkFee(
         }
     }
 
-    XRPAmount const feeDefault = config.FEES.reference_fee;
-
-    auto ledger = app.openLedger().current();
-    // Administrative and identified endpoints are exempt from local fees.
-    XRPAmount const loadFee =
-        scaleFeeLoad(feeDefault, feeTrack, ledger->fees(), isUnlimited(role));
-    XRPAmount fee = loadFee;
-    {
-        auto const metrics = txQ.getMetrics(*ledger);
-        auto const baseFee = ledger->fees().base;
-        auto escalatedFee =
-            toDrops(metrics.openLedgerFeeLevel - FeeLevel64(1), baseFee) + 1;
-        fee = std::max(fee, escalatedFee);
-    }
-
-    auto const limit = [&]() {
-        // Scale fee units to drops:
-        auto const result = mulDiv(feeDefault, mult, div);
-        if (!result)
-            Throw<std::overflow_error>("mulDiv");
-        return *result;
-    }();
-
-    if (fee > limit)
-    {
-        std::stringstream ss;
-        ss << "Fee of " << fee << " exceeds the requested tx limit of "
-           << limit;
-        return RPC::make_error(rpcHIGH_FEE, ss.str());
-    }
-
-    tx[jss::Fee] = fee.jsonClipped();
+    auto feeOrError =
+        getCurrentNetworkFee(role, config, feeTrack, txQ, app, tx, mult, div);
+    if (feeOrError.isMember(jss::error))
+        return feeOrError;
+    tx[jss::Fee] = feeOrError;
```

Comment:
`std::move(feeOrError)`

Reply:
Fixed in 0b818f33287094da7e46370c20269dac49b7a540